common/src/main/java/tachyon/network/protocol/RPCMessageEncoder.java

```diff
@@ -42,9 +42,9 @@ protected void encode(ChannelHandlerContext ctx, RPCMessage in, List<Object> out
     long bodyBytes = 0;
     DataBuffer payload = null;
 
-    if (in.hasPayload()) {
-      payload = in.getPayloadDataBuffer();
-      bodyBytes = payload.getLength();
+    if (in instanceof PayloadMessage) {
+      payload = ((PayloadMessage)in).getPayloadDataBuffer();
+      bodyBytes = (payload == null) ? 0 : payload.getLength();
```

Comment:
I originally thought making the payload be part of a separate interface would be nicer, so then the base RPCMessage wouldn't need payload specific methods (not all messages need payloads). I thought the change would only require a cast, and the rest would be the same.

However, I didn't know that some PayloadMessages could actually have a null payload. I'm not sure if that is expected. In which situations will a PayloadMessage have a null payload?